app/Nova/Travel.php

```diff
@@ -512,19 +569,82 @@ static function (BooleanGroup $field, NovaRequest $request, FormData $formData):
+                    $assignment->envelope->reduce(
+                        // ensure every envelope has a summary PDF on disk
+                        static fn (bool $carry, \App\Models\DocuSignEnvelope $envelope): bool => $carry &&
+                            $envelope->summary_filename !== null &&
+                            // @phan-suppress-next-line PhanPossiblyNullTypeArgumentInternal
+                            file_exists($envelope->summary_filename),
+                        true
+                    ),
+                true
+            )
+        ) {
+            $actions[] = Actions\DownloadDocuSignForms::make()
                 ->canSee(static fn (Request $request): bool => $request->user()->can('view-docusign-envelopes') ||
-                        \App\Models\Travel::where('primary_contact_user_id', $request->user()->id)->exists())
+                    \App\Models\Travel::where('primary_contact_user_id', $request->user()->id)->exists())
                 ->canRun(
                     static fn (NovaRequest $request, AppModelsTravel $travel): bool => $request->user()->can(
                         'view-docusign-envelopes'
                     ) ||
-                            $travel->primaryContact->id === $request->user()->id
-                ),
+                        $travel->primaryContact->id === $request->user()->id
+                );
+        }
 
-            MatrixAirfareSearch::make(),
-        ];
+        if ($trip->status === 'draft' && $trip->needs_airfare_form) {
+            $actions[] = MatrixAirfareSearch::make();
+        }
+
+        if ($trip->status === 'draft' && $request->user()->can('approve-travel')) {
+            if ($request->user()->id === $trip->primary_contact_user_id) {
+                $actions[] = Action::danger(
+                    ReviewTrip::make()->name(),
+                    'You can\'t review this trip because you are the primary contact.'
+                )
+                    ->withoutConfirmation()
+                    ->withoutActionEvents()
+                    ->canRun(static fn (): bool => true);
+            } elseif ($request->user()->id === $trip->created_by_user_id) {
+                $actions[] = Action::danger(
+                    ReviewTrip::make()->name(),
+                    'You can\'t review this trip because you created it.'
+                )
+                    ->withoutConfirmation()
+                    ->withoutActionEvents()
+                    ->canRun(static fn (): bool => true);
+            } elseif ($trip->actions()->where('user_id', '=', $request->user()->id)->exists()) {
+                $actions[] = Action::danger(
+                    ReviewTrip::make()->name(),
+                    'You can\'t review this trip because you have made changes to it.'
+                )
+                    ->withoutConfirmation()
+                    ->withoutActionEvents()
+                    ->canRun(static fn (): bool => true);
```

Comment:
this implies that if there are minor issues to fix, the reviewer can't just fix them, which might be a mildly frustrating thing

Reply:
Yeah, that's intentional. I'm going for a separation of duties here.